File: internal/domain/pending.go
package domain

import (
	"time"

	"github.com/crimsonn/media_pipeline/internal/db/queries"
)

type PendingFileResponse struct {
	ID        int64     `json:"id"`
	FileName  string    `json:"file_name"`
	Status    string    `json:"status"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type PendingEnqueueRequest struct {
	ID                 int64  `json:"id" binding:"required"`
	FileName           string `json:"file_name" binding:"required"`
	TranscodeProfileID int64  `json:"transcode_profile_id" binding:"required"`
}

func ToPendingResponse(pending queries.PendingFile) *PendingFileResponse {
	return &PendingFileResponse{
		ID:        pending.ID,
		FileName:  pending.FileName,
		Status:    pending.Status,
		CreatedAt: pending.CreatedAt.Time,
		UpdatedAt: pending.UpdatedAt.Time,
	}
}

File: web/src/lib/pending.ts
import { api } from '#/lib/api'

/** Mirrors domain.PendingFileResponse (internal/domain/pending.go). */
export interface PendingFile {
  id: number
  file_name: string
  status: string
  created_at: string
  updated_at: string
}

/**
 * Mirrors domain.PendingEnqueueRequest. Both fields use `binding:"required"`
 * on the Go side, so `transcode_profile_id` must be > 0.
 */
export interface PendingEnqueueInput {
  id: number
  file_name: string
  transcode_profile_id: number
}

/**
 * GET /api/v1/pending/files
 * Files dropped into the watch folder that are waiting for a transcode
 * profile to be assigned.
 */
export async function listPendingFiles(): Promise<PendingFile[]> {
  const { data } = await api.get<PendingFile[]>('/pending/files')
  return data ?? []
}

/**
 * POST /api/v1/pending/files
 * Assigns a transcode profile to a pending file and enqueues it for transcoding.
 */
export async function enqueuePendingFile(
  input: PendingEnqueueInput,
): Promise<void> {
  await api.post('/pending/files', input)
}
